q2_mlab/db/tests/test_db.py

```diff
@@ -198,6 +213,108 @@ def test_add_parameters_thrice(self):
 
         fh.close()
 
+    def test_classification_schema(self):
+        engine = create(echo=False)
+
+        param_df = pd.read_sql_table(Parameters.__tablename__, con=engine)
+        self.assertEqual(0, len(param_df))
+
+        parameters = {'max_features_STRING': 'log2'}
+        results = pd.DataFrame([
+            {
+                'CV_IDX': 0, 'RUNTIME': 2.3, 'PROB_CLASS_0': 0.502,
+                'PROB_CLASS_1': 0.498, 'AUPRC': 0.3, 'AUROC': 0.3, 'F1': 0.0
+            },
+            {
+                'CV_IDX': 1, 'RUNTIME': 2.9, 'PROB_CLASS_0': 0.603,
+                'PROB_CLASS_1': 0.397, 'AUPRC': 0.4, 'AUROC': 0.5, 'F1': 0.0
+            },
+        ])
+        dataset = 'FINRISK'
+        algorithm = 'RandomForestClassifier'
+        level = 'MG'
+        target = 'AGE'
+        add(engine=engine, results=results, parameters=parameters,
+            dataset=dataset, algorithm=algorithm, level=level, target=target,
+            artifact_uuid='some-unique-uuid',
+            )
+
+        param_df = pd.read_sql_table(
+            Parameters.__tablename__, con=engine
+        )
+        self.assertEqual(1, len(param_df))
+        score_df = pd.read_sql_table(
+            ClassificationScore.__tablename__, con=engine
+        )
+        self.assertEqual(2, len(score_df))
+
+        # confirm nothing is in regression:
+        score_df = pd.read_sql_table(
+            RegressionScore.__tablename__, con=engine
+        )
+        self.assertEqual(0, len(score_df))
+
+    def test_block_repeated_uuids(self):
+        format_name = "SampleData[Results]"
+
+        results = pd.DataFrame([
+            {'CV_IDX': 0, 'SAMPLE_ID': 'SAMPLE-1', 'Y_PRED': 4, 'Y_TRUE': 4,
+             'RUNTIME': 3.5, 'MAE': 2.4, 'RMSE': 7.24, 'R2': 0.7},
+        ])
+
+        imported_artifact = Artifact.import_data(format_name,
+                                                 results
+                                                 )
+
+        dataset = 'FINRISK'
+        algorithm = 'RandomForestRegressor'
+        level = 'MG'
+        target = 'AGE'
+
+        fh = tempfile.NamedTemporaryFile()
+        db_file = fh.name
+
+        parameters = {'bootstrap': True, 'criterion': 'mse', 'max_depth': 7,
+                      'min_samples_leaf': 0.41, 'min_samples_split': 0.2,
+                      'n_estimators': 100, 'n_jobs': -1, 'random_state': 2020,
+                      }
+
+        for _ in range(3):
+            engine = add_from_qza(
```

Comment:
It looks like you have not called

`q2_mlab.db.maint:create` on `db_file` yet, so you are trying to insert records into an uninitialized database. I believe this is why your tests are failing.

Reply:
If I am not mistaken, this test will still fail after adding a `create` statement, since `add_from_qza` will raise a `ValueError` the second time it tries to insert `imported_artifact`. Which is the behavior we want. Maybe instead call `add_from_qza` once, validate it is in the database, then try call it again inside of a `assertRaises` context.